src/cpu.rs

```diff
@@ -168,12 +173,39 @@ pub(crate) mod arm {
         }
     }
 
+    #[cfg(all(any(target_os = "freebsd"), target_arch = "aarch64"))]
+    pub fn aarch64_mrs_setup() {
+        extern "C" {
+            fn GFp_aarch64_read_isar0() -> u64;
+        }
+
+        #[inline]
+        fn bits_shift(x: u64, high: usize, low: usize) -> u64 {
+            (x >> low) & ((1 << (high - low + 1)) - 1)
+        }
+
+        let mut features = 0;
+        let aa64isar0 = unsafe { GFp_aarch64_read_isar0() };
```

Comment:
How do we know it is safe to execute the `mrs` instruction?

My understanding is that a conservative person should use `getauxval()` (or equivalent) to determine whether it is safe to execute `mrs`. Am I wrong?

Reply:
It's guaranteed to be safe on FreeBSD 12.0 and newer.

`HWCAP` auxval (#1007) has been merged into 12.2, so I guess it's fine now too. (Used to only be available in -CURRENT.)

Reply:
Thanks for the respond. I am really liking the idea of using `getauxval` because then it works just like Linux and we can avoid adding this tiny chunk of C code, so I think we should continue with #1007.